### PR TITLE
Upgrade to support Spring boot 3

### PR DESCRIPTION
**Pull request by Pirayya**
## Describe the changes
- Upgrade Spring boot to version 3.0.0
- Changed Joran Actions to the new way of handling Actions, looked at the implementation Pivotal made in Spring `org.springframework.boot.logging.logback.Spring*`
- Bumped version to 4.0.0-SNAPSHOT

Closes [#279](https://github.com/akkinoc/logback-access-spring-boot-starter/issues/279)

### CODE REVIEW

File: src/main/kotlin/dev/akkinoc/spring/boot/logback/access/LogbackAccessEvent.kt [hunk 1]
```diff
@@ -37,6 +41,10 @@ class LogbackAccessEvent(private var source: LogbackAccessEventSource) : IAccess
         return source.timeStamp
     }
 
+    override fun getSequenceNumber(): Long {
```
Comment (Pirayya) [inline src/main/kotlin/dev/akkinoc/spring/boot/logback/access/LogbackAccessEvent.kt:44]: Not sure if this is necessary

**Comment by akkinoc** [comment]
Thanks for contributing!
Please give me some time to think about it.

**Comment by istvan-derda** [comment]
Hi @akkinoc, is there anything we can help with to get a release out with spring boot 3 support? Happy to get involved :)

**Comment by DarkAtra** [comment]
Hey @akkinoc, could you please give a rough estimate on when we can expect this to be released?
I'm happy to get involved as well if there is anything i can help you with.

**Comment by akkinoc** [comment]
Sorry for the late response.
Thank you for your contribution!

I created a v4 branch and merged your changes there.
I will continue working on the release (#336).

Please wait a little longer until the release.

File: src/main/kotlin/dev/akkinoc/spring/boot/logback/access/LogbackAccessEvent.kt (hunk 1 above)
Comment (akkinoc) [reply src/main/kotlin/dev/akkinoc/spring/boot/logback/access/LogbackAccessEvent.kt:40]: It seems like a new feature has been added in Logback v1.3.
https://logback.qos.ch/manual/configuration.html#sequenceNumberGenerator

I have made corrections here:
https://github.com/akkinoc/logback-access-spring-boot-starter/pull/336/commits/7bc484d70f0816324873491e90aa47effb114782

**Comment by akkinoc** [comment]
Thank you for waiting. I have released v4! 🎉

**Pull request merged**